// config/Prop.js
var esdb = require('easy_db');
var exports = {};
var target = 'dev';

var argv = process.argv;
var kvs = {};
for(var key in argv)
{
    if(key > 1)
    {
        var kv = argv[key].split("=");
        kvs[kv[0]] = kv[1];
    }
}
if(kvs.target)
{
    target = kvs.target;
}

exports.platform = {};
exports.platform.site = {
    hostname: '127.0.0.1',
    port: 9088,
    path: '/mcp-filter/main/interface.htm',
    method: 'POST'
};

//runtime target
exports.target = target;

//config db basic type
var dbs = [{
    config:{'url':'mongodb://127.0.0.1:27017/webgl'},
    type:esdb.prop.dbType.mongodb
},{
    config:{
        'host':'localhost',
        'user':'root',
        'password':'123456',
        'port':3306,
        'database':'webgl'
    },
    type:esdb.prop.dbType.mysql,
    dateToLong:true
},{
    config:{'url':'mongodb://127.0.0.1:27017/webgl'},
    type:esdb.prop.dbType.mongodb
}];
exports.dbs = dbs;

//暂时支持3种密钥来源
exports.digestFromType = {"NONE":0, "DB":1, "CACHE":2, "FIX":3};
exports.digestFromTypeArray = [
    {id:0, code:'NONE', des:"无"},
    {id:1, code:'DB', des:"数据库"},
    {id:2, code:'CACHE', des:"缓存"},
    {id:3, code:'FIX', des:"缓存固定"}
];

//if user hasn't operation in half a hour, the key will be expired.
exports.loginExpiredSeconds = 30*60;

module.exports = exports;
