// control/DemoPageControl.js
var dc = require('dao').dc;

var config = require('config');
var ec = config.ec;

var esut = require('easy_util');
var digestUtil = esut.digestUtil;

var PageControl = function(){};

PageControl.prototype.handle = function(headNode, bodyNode, cb)
{
    console.log(bodyNode);
    var self = this;
    var cmd = headNode.cmd;
    if(self[cmd[1]])
    {
        self[cmd[1]](headNode, bodyNode, cb);
    }
    else
    {
        cb(ec.E4001);
    }
};


PageControl.prototype.sample1 = function(headNode, bodyNode, cb)
{
    var self = this;
    cb(null, {});
};

PageControl.prototype.sample2 = function(headNode, bodyNode, cb)
{
    var self = this;
    cb(null, {});
};

PageControl.prototype.sample3 = function(headNode, bodyNode, cb)
{
    var self = this;
    cb(null, {});
}

PageControl.prototype.sample4 = function(headNode, bodyNode, cb)
{
    var self = this;
    cb(null, {});
}

PageControl.prototype.sample5 = function(headNode, bodyNode, cb)
{
    var self = this;
    cb(null, {});
}

PageControl.prototype.sample6 = function(headNode, bodyNode, cb)
{
    var self = this;
    cb(null, {});
}

/**
 * 正弦函数
 * @param headNode
 * @param bodyNode
 * @param cb
 */
PageControl.prototype.sin = function(headNode, bodyNode, cb)
{
    var self = this;
    cb(null, {});
}

var pageControl = new PageControl();
module.exports = pageControl;
